app/models/event.rb

```diff
@@ -0,0 +1,48 @@
+# == Schema Information
+#
+# Table name: events
+#
+#  id              :bigint           not null, primary key
+#  data            :jsonb
+#  handleable_type :string
+#  name            :string
+#  type            :string
+#  created_at      :datetime         not null
+#  updated_at      :datetime         not null
+#  handleable_id   :bigint
+#
+# Indexes
+#
+#  index_events_on_handleable_type_and_handleable_id  (handleable_type,handleable_id)
+#
+
+class Event < ApplicationRecord
+  belongs_to :handleable, polymorphic: true, optional: true
+  validates :name, :data, presence: true
+
+  class << self
+    def resolve(payload)
+      event = payload[:event]
+      return handle(payload) if handleable?(event)
+
+      EventJob.perform_later(payload, event)
```

Comment:
Aren't we creating the event twice? Here and in `assign_event`.

Reply:
We don't create the Event until it reaches the event service, or if don't handle it